firstvoices/backend/views/membership_views.py

```diff
@@ -40,7 +109,28 @@ class MembershipViewSet(
     """
 
     serializer_class = MembershipDetailSerializer
-    http_method_names = ["get"]
+
+    def initial(self, *args, **kwargs):
+        """Ensures user has permission to perform the requested action."""
+        super().initial(*args, **kwargs)
+
+        if self.request.method.lower() in ["put", "patch", "delete"]:
+            requesting_user = self.request.user
+            membership = Membership.objects.filter(pk=kwargs["pk"]).first()
+
+            # If the membership being changed is that of a Language Admin
+            # ensure that the requesting user is at least staff
+            if membership.role == Role.LANGUAGE_ADMIN:
+                if hasattr(requesting_user, "app_role"):
```

Comment:
A better option here would be to call the permission predicate as a function, so you don't have to worry about checking all the edge cases. (I think this might need additional handling for the AnonymousUser otherwise.) So this could be `if predicates.is_at_least_staff_admin(requesting_user, None)...`